.changes/unreleased/NOTES-20240229-142101.yaml

```diff
@@ -0,0 +1,7 @@
+kind: NOTES
+body: 'all: This Go module has been updated to Go 1.21 per the [Go support policy](https://go.dev/doc/devel/release#policy).
```

Comment:
nit: Should we mention that the `v1.6.0` release was the actual introduction of Go 1.21? Not sure how "backporting" changelogs works for us.

Reply:
Have updated the changelog